portals/publisher/source/src/app/components/Apis/Details/Configuration/components/UpdateWithoutDetails.jsx

```diff
@@ -0,0 +1,261 @@
+            <Typography variant='h6'>{children}</Typography>
+            {onClose ? (
+                <IconButton aria-label='close' className={classes.closeButton} onClick={onClose}>
+                    <CloseIcon />
+                </IconButton>
+            ) : null}
+        </MuiDialogTitle>
+    );
+});
+
+const DialogContent = withStyles((theme) => ({
+    root: {
+        padding: theme.spacing(2),
+    },
+}))(MuiDialogContent);
+
+const DialogActions = withStyles((theme) => ({
+    root: {
+        margin: 0,
+        padding: theme.spacing(1),
+    },
+}))(MuiDialogActions);
+
+/**
+ *
+ * @returns
+ */
+export default function UpdateWithoutDetails(props) {
+    const {
+        classes, api, apiConfig, handleClick, open, handleClose,
+    } = props;
+
+    let availableEndpoint = '';
+    let endpointType = 'PRODUCTION';
+    if (apiConfig.endpointConfig) {
+        if (apiConfig.endpointConfig.production_endpoints) {
+            availableEndpoint = apiConfig.endpointConfig.production_endpoints.url;
+            endpointType = 'PRODUCTION';
+        } else if (apiConfig.endpointConfig.sandbox_endpoints) {
+            availableEndpoint = apiConfig.endpointConfig.sandbox_endpoints.url;
+            endpointType = 'SANDBOX';
+        }
+    } else if (api.advertiseInfo && api.advertiseInfo.apiExternalProductionEndpoint) {
+        availableEndpoint = api.advertiseInfo.apiExternalProductionEndpoint;
+        endpointType = 'PRODUCTION';
+    } else if (api.advertiseInfo && api.advertiseInfo.apiExternalSandboxEndpoint) {
+        availableEndpoint = api.advertiseInfo.apiExternalSandboxEndpoint;
+        endpointType = 'SANDBOX';
+    }
+
+    const [endpointUrl, setEndpointUrl] = useState(availableEndpoint);
+    const [isValidEndpoint, setValidEndpoint] = useState(availableEndpoint !== '');
+    const [availableTiers, setAvailableTiers] = useState([]);
+
+    useEffect(() => {
+        if (apiConfig.policies.length > 0) {
+            setAvailableTiers(apiConfig.policies);
+        } else {
+            const isAsyncAPI = (api.type === 'WS' || api.type === 'WEBSUB' || api.type === 'SSE'
+                || api.type === 'ASYNC');
+            if (isAsyncAPI) {
+                API.asyncAPIPolicies().then((response) => {
+                    if (response.body.list && response.body.list.length > 0) {
+                        setAvailableTiers([response.body.list[0].name]);
+                    }
+                });
+            } else {
+                API.policies('subscription').then((response) => {
+                    if (response.body.list && response.body.list.length > 0) {
+                        setAvailableTiers([response.body.list[0].name]);
+                    }
+                });
+            }
+        }
+    }, [apiConfig]);
+
+    /**
+     * Validate external endpoint
+     *
+     * @param event
+     */
+    const handleOnChangeEndpoint = (event) => {
```

Comment:
Use joi to validate

Reply:
Done with https://github.com/wso2/apim-apps/pull/130/commits/36b280d9d2c55309b71d89935689fff5fbca271b